nomos-blend/message/src/encap/validated.rs

```diff
@@ -18,32 +26,86 @@ pub struct RequiredProofOfSelectionVerificationInputs {
     pub total_membership_size: u64,
 }
 
-/// An incoming encapsulated Blend message whose public header has been
-/// verified.
-///
-/// It can be decapsulated, but before being sent out as-is, it needs to be
-/// converted into its outgoing variant.
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct IncomingEncapsulatedMessageWithValidatedPublicHeader(EncapsulatedMessage);
+/// A received encapsulated message whose public header has been verified.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Hash)]
+pub struct EncapsulatedMessageWithVerifiedPublicHeader {
```

Comment:
```suggestion
/// A encapsulated message whose public header has been verified.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Hash)]
pub struct EncapsulatedMessageWithVerifiedPublicHeader {
```
I think we can modify the comment, because this struct is now used when we create an encapsulated message locally by `new()`